private/runtime/binding-spec.rkt

```diff
@@ -271,16 +280,24 @@
                ([spec specs])
        (simple-expand-internal spec st local-scopes))]
     
-    [(nest pv f inner-spec)
-     (define init-seq (get-pvar st pv))
+    [(nest depth pv f inner-spec)
+     (when (> depth 1)
+       (error "don't know how to handle depth > 1 yet"))
```

Comment:
In the static check case, we shouldn't need `depth` here at all (as commented above) because in the depth = 0 case we use nest-one instead at runtime.

In the dynamic case, there is I think a situation we won't be able to handle without further work on the runtime, and for which we will want a friendly error:

```
(let** [(b:bg ...) e] ...)
#:binding (nest b e)
```

Though I'm not sure that the current depth argument here will be the right thing to help us produce this error, and we could produce the error statically in any case.